retrofit-clients/src/main/java/com/palantir/remoting/retrofit/errors/RetrofitSerializableErrorErrorHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.retrofit.errors;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.remoting.http.errors.SerializableErrorErrorDecoder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import retrofit.ErrorHandler;
+import retrofit.RetrofitError;
+import retrofit.client.Header;
+import retrofit.client.Response;
+
+public enum RetrofitSerializableErrorErrorHandler implements ErrorHandler {
+    INSTANCE;
+
+    @Override
+    public Throwable handleError(RetrofitError cause) {
+        Response response = cause.getResponse();
+
+        Collection<String> contentTypes = ImmutableSet.of();
+        for (Header header : response.getHeaders()) {
+            if (header.getName().equals("Content-Type")) {
+                contentTypes = ImmutableSet.of(header.getValue());
+                break;
+            }
+        }
+
+        InputStream body;
+        try {
+            body = response.getBody().in();
+        } catch (IOException e) {
+            return new RuntimeException("Cannot get input stream from response: " + e.getMessage(), e);
+        }
+        return SerializableErrorErrorDecoder.INSTANCE.decode(contentTypes, response.getStatus(), response.getReason(),
```

Comment:
well, the point of the error decoder was to never throw an exception but to always return one....do you still think we should wrap in try/catch though? (the try block above is actually specific to getting the input stream currently....you propose we widen the responsibility of the try/catch?)